client/src/features/visualizer-threejs/ConfigControls.tsx

```diff
@@ -0,0 +1,245 @@
+        const input = inputs.find((input) => input.key === key);
+        if (!input) return;
+
+        if (!val) {
+            setErrors((prevErrors) => ({ ...prevErrors, [key]: "Value is required" }));
+            setVisualizerConfigValues((prevState) => ({ ...prevState, [key]: "" }));
+            return;
+        }
+
+        const numericValue = Number(val);
+        if (numericValue < input.min || numericValue > input.max) {
+            setErrors((prevErrors) => ({ ...prevErrors, [key]: `Value must be between ${input.min} and ${input.max}` }));
+        } else {
+            setErrors((prevErrors) => ({ ...prevErrors, [key]: "" }));
+        }
+
+        setVisualizerConfigValues((prevState) => ({ ...prevState, [key]: numericValue }));
+    };
+
+    if (!features.controlsVisualiserEnabled) {
+        return null;
+    }
+
+    return (
+        <div className={"controls-container"}>
+            <div className="controls__list">
+                {inputs.map((i) => {
+                    return (
+                        <div key={i.key} className="controls__item">
+                            <label>{i.label}</label>
+                            <input
+                                type="number"
+                                value={visualizerConfigValues[i.key]}
+                                onChange={(e) => handleChange(i.key, e.target.value)}
+                            />
+                            {!!errors[i.key] && <div className={"controls__error"}>{errors[i.key]}</div>}
+                        </div>
+                    );
+                })}
+            </div>
+
+            <div className="controls__actions">
+                <button type={"button"} onClick={handleApply}>
+                    Apply
+                </button>
+                {showResetButton && (
+                    <button
+                        onClick={() => {
+                            localStorage.removeItem(VISUALIZER_CONFIG_LOCAL_STORAGE_KEY);
+                            setShowResetButton(false);
+                            location.reload();
+                        }}
+                    >
+                        Reset
+                    </button>
+                )}
+            </div>
+
+            <div style={{ marginTop: "16px" }}>
+                <label style={{ display: "block" }}>Zoom</label>
+                <input
+                    value={localZoom === undefined ? "" : localZoom}
+                    onChange={(e) => {
+                        const input = e.target.value;
+
+                        if (!input) {
+                            setLocalZoom(undefined);
+                            return;
+                        }
+
+                        const numberRegExp = /^-?\d+(\.|\.\d*|\d*)?$/;
+                        if (numberRegExp.test(input)) {
+                            if (input.endsWith(".")) {
+                                setLocalZoom(input as any);
+                            } else {
```

Comment:
Is this condition necessary? In the console `parseFloat("2.") === parseFloat("2")`